src/lambda/vue/samInvoke.ts

```diff
@@ -397,13 +371,11 @@ async function saveLaunchConfig(config: AwsSamDebuggerConfiguration): Promise<vo
  * @param config Config to invoke
  */
 async function invokeLaunchConfig(config: AwsSamDebuggerConfiguration, context: ExtContext): Promise<void> {
-    const finalConfig = finalizeConfig(config, 'Editor-Created Debug Config')
-
+    const finalConfig = finalizeConfig(resolveWorkspaceFolderVariable(undefined, config), 'Editor-Created Debug Config')
     const targetUri = getUriFromLaunchConfig(finalConfig)
-
     const folder = targetUri ? vscode.workspace.getWorkspaceFolder(targetUri) : undefined
 
-    await new SamDebugConfigProvider(context).resolveDebugConfiguration(folder, finalConfig)
+    await new SamDebugConfigProvider(context).resolveDebugConfigurationWithSubstitutedVariables(folder, finalConfig)
```

Comment:
Going to swap this out for a `startDebugging` call since that does all of the resolution steps properly (just realized that that always executed the full resolution chain). However, we still need to infer the folder since that call requires that you provide the active workspace folder.

Reply:
Does this avoid the extra `doTraverseAndReplace()` function?

Reply:
I could refactor into a smaller function that just infers a workspace folder. We need a workspace folder for the startDebugging call, and the only way to have that is to get a full path. In that case, we'd look for the first workspaceFolder var and kick out of the entire recursive stack once we're able to infer a single value.